app/src/main/java/org/simple/clinic/teleconsultlog/teleconsultrecord/screen/TeleconsultRecordEffectHandler.kt

```diff
@@ -36,9 +43,35 @@ class TeleconsultRecordEffectHandler @AssistedInject constructor(
         .addTransformer(LoadPatientDetails::class.java, loadPatientDetails())
         .addAction(ShowTeleconsultNotRecordedWarning::class.java, uiActions::showTeleconsultNotRecordedWarning, schedulersProvider.ui())
         .addTransformer(ValidateTeleconsultRecord::class.java, validateTeleconsultRecord())
+        .addTransformer(ClonePatientPrescriptions::class.java, clonePatientPrescriptions())
         .build()
   }
 
+  private fun clonePatientPrescriptions(): ObservableTransformer<ClonePatientPrescriptions, TeleconsultRecordEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .map { effect -> prescriptionRepository.newestPrescriptionsForPatientImmediate(effect.patientUuid) to effect }
+          .doOnNext { (prescriptions, _) -> prescriptionRepository.softDeletePrescriptions(prescriptions) }
```

Comment:
Yep, filtering prescriptions that are empty before soft deleting or cloning them. Is that fine?

```kt
filter { (prescriptions, _) -> prescriptions.isNotEmpty() }
```

Reply:
This should work, one concern is whether we are dependent on the event emitted from this effect handler to proceed with business logic.

Reply:
I don't understand the concern 😅

Reply:
Are we dependent on the `PrescriptionCloned` event that is emitted from this effect to do some work in the `update` function?

If yes, then this filter will not work because the event will never be emitted which will stop processing of important business logic.

Reply:
Ah okay, yeah we need to navigate to teleconsult success screen after cloning is completed. Is there a way to send a event after the filter or is there other way I can achieve this?

Reply:
We can still send the `PrescriptionCloned` event, just not call any of the methods if there are no prescriptions available.